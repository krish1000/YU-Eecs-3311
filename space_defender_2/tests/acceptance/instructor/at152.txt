--Test Recall into a projectile.
toggle_debug_mode

play(5,28,1,101,101,101,101)
setup_select(3)
setup_next(3)
setup_select(1)
setup_next(1)
setup_next(1)

move(B,1)
move(B,2)
move(B,1)
pass
move(B,2)
special
