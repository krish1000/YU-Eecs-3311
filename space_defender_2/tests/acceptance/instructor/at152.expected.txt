  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,28,1,101,101,101,101)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(3)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Snipe
->setup_next(3)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:setup summary, debug, ok
  Weapon Selected:Snipe
  Armour Selected:None
  Engine Selected:Standard
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:160/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[C,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:158/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[B,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [B,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Fighter(id:1) spawns at location [A,28].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    B S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,2)
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:158/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[B,2]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,25]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,28]
  Projectile:
    [-1,<]->damage:20, move:3, location:[A,24]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [B,1] -> [B,2]
  Enemy Action:
    A Fighter(id:1) moves: [A,28] -> [A,25]
      A enemy projectile(id:-1) spawns at location [A,24].
  Natural Enemy Spawn:
    A Fighter(id:2) spawns at location [A,28].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  _  F
    B _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,1)
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:158/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[B,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,22]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,25]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[C,28]
  Projectile:
    [-2,<]->damage:40, move:3, location:[A,21]
    [-3,<]->damage:20, move:3, location:[A,24]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,24] -> [A,21]
  Starfighter Action:
    The Starfighter(id:0) moves: [B,2] -> [B,1]
  Enemy Action:
    A Fighter(id:1) moves: [A,25] -> [A,22]
      A enemy projectile(id:-2) spawns at location [A,21].
      The projectile collides with enemy projectile(id:-1) at location [A,21], combining damage.
    A Fighter(id:2) moves: [A,28] -> [A,25]
      A enemy projectile(id:-3) spawns at location [A,24].
  Natural Enemy Spawn:
    A Fighter(id:3) spawns at location [C,28].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  <  F  _  _  _
    B S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:160/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[B,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,16]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,19]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[C,22]
    [4,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,28]
  Projectile:
    [-4,<]->damage:100, move:10, location:[A,15]
    [-5,<]->damage:100, move:10, location:[A,18]
    [-6,<]->damage:100, move:10, location:[C,21]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-2) moves: [A,21] -> [A,18]
    A enemy projectile(id:-3) moves: [A,24] -> [A,22]
      The projectile collides with Fighter(id:1) at location [A,22], healing 20 damage.
  Starfighter Action:
    The Starfighter(id:0) passes at location [B,1], doubling regen rate.
  Enemy Action:
    A Fighter(id:1) moves: [A,22] -> [A,16]
      The Fighter collides with enemy projectile(id:-2) at location [A,18], healing 40 damage.
      A enemy projectile(id:-4) spawns at location [A,15].
    A Fighter(id:2) moves: [A,25] -> [A,19]
      A enemy projectile(id:-5) spawns at location [A,18].
    A Fighter(id:3) moves: [C,28] -> [C,22]
      A enemy projectile(id:-6) spawns at location [C,21].
  Natural Enemy Spawn:
    A Fighter(id:4) spawns at location [B,28].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  <  F  _  _  _  _  _  _  _  _  _
    B S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,2)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:158/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[B,2]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,13]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,16]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[C,19]
    [4,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,25]
    [5,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[D,28]
  Projectile:
    [-4,<]->damage:100, move:10, location:[A,5]
    [-6,<]->damage:100, move:10, location:[C,11]
    [-7,<]->damage:20, move:3, location:[A,12]
    [-8,<]->damage:20, move:3, location:[A,15]
    [-9,<]->damage:20, move:3, location:[C,18]
    [-10,<]->damage:20, move:3, location:[B,24]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-4) moves: [A,15] -> [A,5]
    A enemy projectile(id:-5) moves: [A,18] -> [A,16]
      The projectile collides with Fighter(id:1) at location [A,16], healing 100 damage.
    A enemy projectile(id:-6) moves: [C,21] -> [C,11]
  Starfighter Action:
    The Starfighter(id:0) moves: [B,1] -> [B,2]
  Enemy Action:
    A Fighter(id:1) moves: [A,16] -> [A,13]
      A enemy projectile(id:-7) spawns at location [A,12].
    A Fighter(id:2) moves: [A,19] -> [A,16]
      A enemy projectile(id:-8) spawns at location [A,15].
    A Fighter(id:3) moves: [C,22] -> [C,19]
      A enemy projectile(id:-9) spawns at location [C,18].
    A Fighter(id:4) moves: [B,28] -> [B,25]
      A enemy projectile(id:-10) spawns at location [B,24].
  Natural Enemy Spawn:
    A Fighter(id:5) spawns at location [D,28].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  <  _  _  _  _  _  _  <  F  _  <  F  _  _  _  _  _  _  _  _  _  _  _  _
    B _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  _  _
    C _  _  _  _  _  _  _  _  _  _  <  _  _  _  _  _  _  <  F  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->special
  state:not started, debug, ok
  Starfighter:
    [0,S]->health:0/60, energy:110/160, Regen:1/7, Armour:1, Vision:22, Move:12, Move Cost:2, location:[C,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,13]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,16]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[C,19]
    [4,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[B,25]
    [5,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[D,28]
  Projectile:
    [-7,<]->damage:20, move:3, location:[A,9]
    [-9,<]->damage:20, move:3, location:[C,15]
    [-10,<]->damage:20, move:3, location:[B,21]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-4) moves: [A,5] -> out of board
    A enemy projectile(id:-6) moves: [C,11] -> [C,1]
    A enemy projectile(id:-7) moves: [A,12] -> [A,9]
    A enemy projectile(id:-8) moves: [A,15] -> [A,13]
      The projectile collides with Fighter(id:1) at location [A,13], healing 20 damage.
    A enemy projectile(id:-9) moves: [C,18] -> [C,15]
    A enemy projectile(id:-10) moves: [B,24] -> [B,21]
  Starfighter Action:
    The Starfighter(id:0) uses special, teleporting to: [C,1]
      The Starfighter collides with enemy projectile(id:-6) at location [C,1], taking 99 damage.
      The Starfighter at location [C,1] has been destroyed.
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
    A _  _  _  _  _  _  _  _  <  _  _  _  F  _  _  F  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  _  _  _  F  _  _  _
    C X  _  _  _  _  _  _  _  _  _  _  _  _  _  <  _  _  _  F  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
  The game is over. Better luck next time!
